cg/apps/housekeeper/hk.py

```diff
@@ -443,3 +444,35 @@ def get_non_archived_spring_path_and_bundle_name(self) -> List[Tuple[str, str]]:
             (file.version.bundle.name, file.path)
             for file in self.get_all_non_archived_spring_files()
         ]
+
+    def set_archive_file_retrieved_at(self, file_id: int, retrieval_task_id: int) -> None:
+        file: File = self._store.get_file_by_id(file_id)
+        archive: Archive = file.archive
+        if archive.retrieval_task_id != retrieval_task_id:
+            raise ValueError(
+                f"Retrieval task id did not match database entry. Given task id was {retrieval_task_id}, "
+                f"while retrieval task id in Housekeeper is {archive.retrieval_task_id}."
+            )
+        self._store.update_retrieval_time_stamp(archive=archive)
+        self.commit()
+
+    def set_archive_file_archived_at(self, file_id: int, archiving_task_id: int) -> None:
```

Comment:
Will these ever be used? I only see us using the functions intended for "all" files with the specified task id.